apps/web/src/components/PoolCandidatesTable/JobPlacementDialog.tsx

```diff
@@ -0,0 +1,317 @@
+          })}
+        >
+          {intl.formatMessage({
+            defaultMessage: "Job placement",
+            id: "W3NvhG",
+            description: "Heading job placement dialog",
+          })}
+        </Dialog.Header>
+        <Dialog.Body>
+          <FormProvider {...methods}>
+            <form onSubmit={handleSubmit(handleFormSubmit)}>
+              <div data-h2-display="base(grid)" data-h2-gap="base(x1)">
+                <RadioGroup
+                  idPrefix="placementType"
+                  name="placementType"
+                  legend={intl.formatMessage({
+                    defaultMessage: "Job placement status",
+                    id: "dpO8Va",
+                    description: "Label for the job placement status field",
+                  })}
+                  items={placementTypeOptions}
+                />
+                {isPlaced && (
+                  <Select
+                    id="placedDepartment"
+                    name="placedDepartment"
+                    label={intl.formatMessage({
+                      defaultMessage: "Placed department",
+                      id: "G8JoCN",
+                      description: "Label for the placed department field",
+                    })}
+                    nullSelection={intl.formatMessage({
+                      defaultMessage: "Select a department",
+                      id: "y827h2",
+                      description:
+                        "Null selection for department select input in the request form.",
+                    })}
+                    options={objectsToSortedOptions([...departments], intl)}
+                    rules={{
+                      required: intl.formatMessage(errorMessages.required),
+                    }}
+                  />
+                )}
+              </div>
+              <Dialog.Footer data-h2-justify-content="base(flex-start)">
+                <Submit
+                  text={intl.formatMessage({
+                    defaultMessage: "Save decision",
+                    id: "DyHuZi",
+                    description: "Submit message for job placement dialog",
+                  })}
+                  color="primary"
+                  mode="solid"
+                />
+                <Dialog.Close>
+                  <Button color="tertiary" mode="inline">
+                    {intl.formatMessage(commonMessages.cancel)}
+                  </Button>
+                </Dialog.Close>
+              </Dialog.Footer>
+            </form>
+          </FormProvider>
+        </Dialog.Body>
+      </Dialog.Content>
+    </Dialog.Root>
+  );
+};
+
+export function jobPlacementDialogAccessor(
+  poolCandidateId: string,
+  departments: Department[],
+  status?: Maybe<PoolCandidateStatus>,
+  placedDepartment?: Maybe<Department>,
+) {
+  const jobPlacementDialogQuery = makeFragmentData(
+    { id: poolCandidateId, status, placedDepartment },
+    JobPlacementDialog_Fragment,
+  );
```

Comment:
We may need to look into this. It shouldn't be necessary to do this. You should be able to just forward on the result of the query here. The args should be typed similar to the dialog props:

```tsx
function jobPlacementDialogAccessor(
  jobPlacementDialogQuery: FragmentType<typeof JobPlacementDialog_Fragment>;
  departments: Department[];
) {}
```

Reply:
Alright was forced to cast it within the props like we discussed. Fixed by [0fef170](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10216/commits/0fef17082e3e9bc0eaae18dcc701dee4bd5767ef)